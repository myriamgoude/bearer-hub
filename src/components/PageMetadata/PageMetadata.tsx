/*
  Helper compoment for pages to set common meta
*/
import * as React from 'react'
import Helmet from 'react-helmet'
import { graphql, StaticQuery } from 'gatsby'

interface IPageMetadataProps {
  title: string
  description?: string | null
  image?: string | null
}

interface IPageMetadataTagsProps extends IPageMetadataProps {
  siteUrl: string
}
const query = graphql`
  query SiteMetaUrl {
    site {
      siteMetadata {
        siteUrl
      }
    }
  }
`
export const PageMetadataTags: React.FunctionComponent<IPageMetadataTagsProps> = ({
  title,
  description,
  image,
  siteUrl
}) => {
  const allDescriptions = description
    ? [{ name: 'description', content: description }, { name: 'twitter:description', content: description }]
    : []

  const fullQualifiedImage = `${siteUrl}${image}`
  const allImages = image ? [{ name: 'twitter:image', content: fullQualifiedImage }] : []

  const allTitles = [{ name: 'twitter:title', content: title }]
  return (
    <Helmet title={title} meta={[...allTitles, ...allDescriptions, ...allImages]}>
      <meta property="og:title" content={title} />
      {description ? <meta property="og:description" content={description} /> : null}
      {image ? <meta property="og:image" content={fullQualifiedImage} /> : null}
    </Helmet>
  )
}

const PageMetadata: React.SFC<IPageMetadataProps> = ({ title, description, image }) => {
  return (
    <StaticQuery
      query={query}
      render={({
        site: {
          siteMetadata: { siteUrl }
        }
      }) => {
        return <PageMetadataTags title={title} description={description} siteUrl={siteUrl} image={image} />
      }}
    />
  )
}
export default PageMetadata
