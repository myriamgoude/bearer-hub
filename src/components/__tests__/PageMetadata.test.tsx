import React from 'react'
import * as ShallowRenderer from 'react-test-renderer/shallow'
import { PageMetadataTags } from '../PageMetadata/PageMetadata'

describe('PageMetadata', () => {
  const title = 'My Page'
  const desc = 'My description'
  const img = 'logo.png'
  const siteUrl = 'http://www.bearer.sh/'
  it('renders when given only a title', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<PageMetadataTags siteUrl={siteUrl} title={title} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders title, description', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<PageMetadataTags siteUrl={siteUrl} title={title} description={desc} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders title, description, image', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<PageMetadataTags siteUrl={siteUrl} title={title} description={desc} image={img} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
