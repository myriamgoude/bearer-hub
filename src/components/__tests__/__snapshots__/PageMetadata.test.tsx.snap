// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMetadata renders title, description 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  meta={
    Array [
      Object {
        "content": "My Page",
        "name": "twitter:title",
      },
      Object {
        "content": "My description",
        "name": "description",
      },
      Object {
        "content": "My description",
        "name": "twitter:description",
      },
    ]
  }
  title="My Page"
>
  <meta
    content="My Page"
    property="og:title"
  />
  <meta
    content="My description"
    property="og:description"
  />
</HelmetWrapper>
`;

exports[`PageMetadata renders title, description, image 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  meta={
    Array [
      Object {
        "content": "My Page",
        "name": "twitter:title",
      },
      Object {
        "content": "My description",
        "name": "description",
      },
      Object {
        "content": "My description",
        "name": "twitter:description",
      },
      Object {
        "content": "http://www.bearer.sh/logo.png",
        "name": "twitter:image",
      },
    ]
  }
  title="My Page"
>
  <meta
    content="My Page"
    property="og:title"
  />
  <meta
    content="My description"
    property="og:description"
  />
  <meta
    content="http://www.bearer.sh/logo.png"
    property="og:image"
  />
</HelmetWrapper>
`;

exports[`PageMetadata renders when given only a title 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  meta={
    Array [
      Object {
        "content": "My Page",
        "name": "twitter:title",
      },
    ]
  }
  title="My Page"
>
  <meta
    content="My Page"
    property="og:title"
  />
</HelmetWrapper>
`;
